src/Network/Transport/TCP.hs

```diff
@@ -517,20 +519,25 @@ data TransportInternals = TransportInternals
 --------------------------------------------------------------------------------
 
 -- | Create a TCP transport
-createTransport :: N.HostName
-                -> N.ServiceName
+createTransport :: N.HostName    -- ^ Bind host name.
+                -> N.ServiceName -- ^ Bind port.
+                -> (N.ServiceName -> (N.HostName, N.ServiceName))
+                   -- ^ External address host name and port, computed from the
+                   --   actual bind port (must be lazy in it).
```

Comment:
Can we go the same way as in https://github.com/avieth/network-transport-tcp/commit/2909c76088658c3b7f39a19c29b24dbcbcf45f15 in that commit we allow to set port that was given to us by the OS.